.github/workflows/build.yaml

```diff
@@ -0,0 +1,19 @@
+name: build
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    paths:
+      - '**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '~1.17'
+      - run: make build
```

Comment:
@njhale I think this is fine for now, at least we can determine quickly whether we'd regressed in being able to compile the provisioner binary, but we may want to move towards a build and unit testing workflow where we actually build the container image -> publish locally -> unit tests are run using that container image?

Reply:
> unit tests are run using that container image?

we normally create sparse images that don't contain source/test files. we could always compile the test binary, but I don't really see a reason to do that.

running `make unit-test` against the source in an action seems most natural here